src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -49,41 +52,125 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>) {
   function updateFunctionalComponent(current, workInProgress) {
     var fn = workInProgress.type;
     var props = workInProgress.pendingProps;
-    console.log('update fn:', fn.name);
     var nextChildren = fn(props);
     reconcileChildren(current, workInProgress, nextChildren);
     workInProgress.pendingWorkPriority = NoWork;
   }
 
-  function updateHostComponent(current, workInProgress) {
-    console.log(
-      'host component', workInProgress.type, 
-      typeof workInProgress.pendingProps.children === 'string' ? workInProgress.pendingProps.children : ''
-    );
+  function updateFiber(fiber: Fiber, state: any, priorityLevel : PriorityLevel): void {
+    const { scheduleLowPriWork } = getScheduler();
+    fiber.pendingState = state;
+
+    while (true) {
+      if (fiber.pendingWorkPriority === NoWork ||
+          fiber.pendingWorkPriority >= priorityLevel) {
+        fiber.pendingWorkPriority = priorityLevel;
+      }
+      // Duck type root
+      if (fiber.stateNode && fiber.stateNode.containerInfo) {
+        const root : FiberRoot = (fiber.stateNode : any);
+        scheduleLowPriWork(root, priorityLevel);
+        return;
+      }
+      if (!fiber.return) {
+        throw new Error('No root!');
+      }
+      fiber = fiber.return;
+    }
+  }
+
+  // Class component state updater
+  const updater = {
+    enqueueSetState(instance, partialState) {
+      const fiber = instance._fiber;
+
+      const prevState = fiber.pendingState || fiber.memoizedState;
```

Comment:
Yea, that's why this will need to become a queue of state transitions for those cases.

You may also need to be able to schedule different priority level state transitions on the *same* component. In that case the queue can be reordered. We can wait on that though. Not sure it is worth the trouble.

Can `pendingState` just use the partial state and that way you can do the merge at the time you apply it?